sklearn/covariance/tests/test_graphical_lasso.py

```diff
@@ -85,6 +85,23 @@ def test_graphical_lasso_iris():
         assert_array_almost_equal(icov, icov_R)
 
 
+def test_graph_lasso_2D():
+    # Hard-coded solution from Python skggm package
+    # obtained by calling `quic(emp_cov, lam=.1, tol=1e-8)`
+    cov_skggm = np.array([[3.09550269, 1.186972],
+                         [1.186972, 0.57713289]])
+
+    icov_skggm = np.array([[1.52836773, -3.14334831],
+                          [-3.14334831,  8.19753385]])
+    X = datasets.load_iris().data[:, 2:]
+    emp_cov = empirical_covariance(X)
+    for method in ('cd', 'lars'):
+        cov, icov = graphical_lasso(emp_cov, alpha=.1, return_costs=False,
+                                    mode=method)
```

Comment:
to make the test more explicit, we could do

```python
emp_cov_copy = emp_cov.copy()
cov, icov = graphical_lasso(emp_cov_copy, ...)
... # the assertions you currently have, but also
assert_array_almost_equal(emp_cov, emp_cov_copy) 
```

which should fail with the old version, right?

Reply:
Nope, because inside `graphical_lasso`, `emp_cov` is first copied to the variable `covariance_`

Reply:
Ah, you're right, great point. This test is great in this case; it seems there's no better way to test for this issue.